src/services/redeem_and_unstake/AcceptRedeemRequestService.ts

```diff
@@ -0,0 +1,337 @@
+  public constructor(
+    repos: Repositories,
+    web3: Web3,
+    redeemPoolAddress: string,
+    auxiliaryWorkerAddress: string,
+  ) {
+    super();
+
+    this.web3 = web3;
+    this.messageTransferRequestRepository = repos.messageTransferRequestRepository;
+    this.messageRepository = repos.messageRepository;
+    this.redeemPoolAddress = redeemPoolAddress;
+    this.auxiliaryWorkerAddress = auxiliaryWorkerAddress;
+  }
+
+  /**
+   * This method react on changes in MessageTransferRequest model.
+   * @param redeemRequests Array of redeem requests.
+   */
+  public async update(redeemRequests: MessageTransferRequest[]): Promise<void> {
+    Logger.debug('Accept redeem request service invoked');
+    const nonAcceptedRedeemRequests = redeemRequests.filter(
+      (redeemRequest: MessageTransferRequest): boolean =>
+        (redeemRequest.requestType === RequestType.Redeem) && !redeemRequest.messageHash,
+    );
+
+    await this.acceptRedeemRequests(nonAcceptedRedeemRequests);
+  }
+
+  /**
+   * Generates and returns hashlock and secret.
+   */
+  public static generateSecret(): { secret: string; hashLock: string } {
+    const secret = Web3Utils.randomHex(32);
+    const hashLock = Web3Utils.keccak256(secret);
+
+    return {
+      secret,
+      hashLock,
+    };
+  }
+
+
+  /* Private Functions */
+
+  /**
+   * Collects acceptRedeemRequest transaction promise and executed them.
+   *
+   * @param redeemRequests Array of redeem requests.
+   */
+  private async acceptRedeemRequests(redeemRequests: MessageTransferRequest[]): Promise<void> {
+    const redeemRequestPromises = [];
+    for (let i = 0; i < redeemRequests.length; i += 1) {
+      redeemRequestPromises.push(
+        this.acceptRedeemRequest(redeemRequests[i]).catch((error) => {
+          Logger.error('acceptRedeemRequestService error', error);
+        }),
+      );
+    }
+
+    await Promise.all(redeemRequestPromises);
+  }
+
+  /**
+   * It sends accept redeem transaction and updates messageHash in MessageTransferRequest
+   * repository.
+   * @param redeemRequest Redeem request object.
+   */
+  private async acceptRedeemRequest(redeemRequest: MessageTransferRequest): Promise<void> {
+    const { secret, hashLock } = AcceptRedeemRequestService.generateSecret();
+
+    const transactionHash = await this.sendAcceptRedeemRequestTransaction(
+      redeemRequest, hashLock,
+    );
+    Logger.info(`Accept redeem request transaction hash ${transactionHash}`);
+    const messageHash = await this.createMessageInRepository(
+      redeemRequest, secret, hashLock,
+    );
+
+    await this.updateMessageHash(
+      redeemRequest.requestHash,
+      messageHash,
+      redeemRequest.blockNumber,
+    );
+  }
+
+  /**
+   * Fetches bounty amount from contract and returns bounty amount.
+   *
+   * @param redeemRequest Redeem request object.
+   * @return {string} Bounty amount.
+   */
+  private async getBountyAmount(redeemRequest: MessageTransferRequest): Promise<string> {
```

Comment:
we have bounty in gateway table in facilitator db, should use from it

Reply:
Good point.
However . Bounty can be changed in Gateway/CoGateway repository. It's always good to read the latest value from blockchain as final source of truth.
We will need bounty value from db while implementing bounty hunter. We can evaluate at that time whether we need bounty field or not.